src/main/java/org/betonquest/betonquest/objectives/CommandObjective.java

```diff
@@ -0,0 +1,76 @@
+package org.betonquest.betonquest.objectives;
+
+import org.betonquest.betonquest.BetonQuest;
+import org.betonquest.betonquest.Instruction;
+import org.betonquest.betonquest.api.Objective;
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.betonquest.betonquest.utils.PlayerConverter;
+import org.bukkit.Bukkit;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerCommandPreprocessEvent;
+
+import java.util.List;
+
+/**
+ * Requires the player to execute a specific command.
+ */
+@SuppressWarnings("PMD.CommentRequired")
+public class CommandObjective extends Objective implements Listener {
+
+    private final String command;
+    private final List<String> commandVariables;
+
+    public CommandObjective(final Instruction instruction) throws InstructionParseException {
+        super(instruction);
+        template = ObjectiveData.class;
+        command = instruction.next().replaceAll("_", " ");
+        commandVariables = BetonQuest.resolveVariables(command);
+    }
+
+    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+    public void onCommand(final PlayerCommandPreprocessEvent event) {
```

Comment:
if you use `AsyncPlayerChatEvent` here, it solves multiple problems.
1. There are plugins, that do not register commands at all, the only listen to the `AsyncPlayerChatEvent` cancel it and interpret it as a command. The most plugins like that are command creation plugins.
2. You are able to check for the execution of commands, that are no known commands. With this, it would be possible, to create commands in BQ directly.

And one more thing:  As a result of 1, you also want the event priority `LOWEST`, because we dont do critical shit, but some other plugins can use a LOWEST priority cancel the event, and we are not able anymore to notice the event.

Reply:
I don't believe I need to use `AsyncPlayerChatEvent`. I will verify, but I believe it calls the `PlayerCommandPreprocessEvent` just if it has a slash at the beginning of the line. It doesn't matter if the command is real and registered, tried with `/hello` and my event still fired. I agree that this should be changed to lowest (I would say monitor, but you want me to cancel the event). I also don't think that ignoreCancelled should be true, what do you think?

Reply:
Also, if you are `LOWEST` and another plugin is `LOWEST`, you are _still_ not guaranteed to be run afterward. That's my reasoning for `MONITOR`.

Reply:
The main reason for `AsyncPlayerChatEvent` are plugins that allow you to create commands. I think Denizen has this, also CMI, and i know there are a lot more, use this event, to listen for not real commands. And if you use `PlayerCommandPreprocessEvent` this objective does not work with that plugins/commands.

Keep LOWEST, we know that other plugins can also have lowest, and there is no guarantee for the order, but since we want to cancel it we are not allowed to use MONITOR.

Reply:
Consider the case where I have a permission plugin that governs if I can run the command. True, that is likely highest, but we really don't know. I should not ignore cancelled commands, and I should only execute if the command is allowed to run (not governed by AsyncPlayerChatEvent!). I have to run the command in a specific location. This requires that it be managed through the PlayerCommandPreprocessEvent.